common/vm/vm.go

```diff
@@ -0,0 +1,69 @@
+package vm
+
+import (
+	"github.com/DataDog/test-infra-definitions/command"
+	"github.com/DataDog/test-infra-definitions/common/agentinstall"
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/common/os"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func InitVM(
+	ctx *pulumi.Context,
+	env config.Environment,
+	instanceIP pulumi.StringInput,
+	os os.OS,
+	optionalAgentInstallParams *agentinstall.Params,
+) (*command.Runner, error) {
+	connection, runner, err := createRunner(ctx, env, instanceIP, os.GetSSHUser())
+	if err != nil {
+		return nil, err
+	}
+
+	if optionalAgentInstallParams != nil {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Does this mean that if `optionalAgentInstallParams` is `nil` we don't install the agent ?

Reply:
Yes

Reply:
> The parameter is unclear

Can you elaborate what is not clear? `optionalAgentInstallParams` is an optional field and when it is not null contains the information for installing the Agent. This code seems straightforward to understand. Did I miss something?

> we also have a ConfigMap parameter for that.

I have updated the code in https://github.com/DataDog/test-infra-definitions/pull/25 (https://github.com/DataDog/test-infra-definitions/pull/25/commits/1d1628008215700cac50ebac0bd0653261be39bc) to limit the code conflict.
Note: This code was introduced in a previous PR.

Reply:
> Does this mean that if optionalAgentInstallParams is nil we don't install the agent ?

The fact that giving no install params deactivate Agent installation is not clear. The name of parameter suggests that indeed it's optional and you **can** give install params but not that not giving any **will** deactivate Agent installation.

I believe the choice to install/uninstall should still be governed with the ConfigMap parameter.